src/leveled_sst.erl

```diff
@@ -1395,41 +1394,30 @@ fetch(LedgerKey, Hash,
 %% Fetch pointers to the slots the SST file covered by a given key range.
 fetch_range(StartKey, EndKey, Summary, FilterFun, true) ->
     {Slots, RTrim} =
-        lookup_slots(
-            StartKey,
-            EndKey,
-            Summary#summary.index,
-            FilterFun),
+        lookup_slots(StartKey, EndKey, Summary#summary.index, FilterFun),
     Self = self(),
-    SL = length(Slots),
-    case SL of
-        1 ->
-            [Slot] = Slots,
-            case RTrim of
-                true ->
-                    [{pointer, Self, Slot, StartKey, EndKey}];
-                false ->
-                    [{pointer, Self, Slot, StartKey, all}]
-            end;
-        N ->
-            {LSlot, MidSlots, RSlot} =
-                {hd(Slots), lists:sublist(Slots, 2, N - 2), lists:last(Slots)},
+    case {Slots, RTrim} of
+        {[Slot], true} ->
+            [{pointer, Self, Slot, StartKey, EndKey}];
+        {[Slot], false} ->
+            [{pointer, Self, Slot, StartKey, all}];
+        {[Hd|Rest], true} ->
+            {MidSlots, [Tl]} = lists:split(length(Rest) - 1, Rest),
```

Comment:
`[Tl]` is kind of misleading here... rather `[Last]` ?

Verified that this works even for a list with only 2 slots. `[Tl]` then is the second slot and `MidSlots` is empty.